proposals/hasCrossSiteAncestor.md

```diff
@@ -0,0 +1,122 @@
+# Proposal: Update Extensions to interact with Partitioned Cookies containing Cross-Site Ancestor Chain Bit values
+
+**Summary**
+
+Allow extensions to utilize a hasCrossSiteAncestor boolean value when interacting with partitioned cookies that include a cross-site ancestor chain bit in their partiton key.
+
+#### Background information: Description of Cross-site Ancestor Chain Bit in partitioned cookies
+
+The cross-site ancestor chain bit is a component of the cookie partition key that is set by the browser when keys are created. If the bit indicates true, it means the cookie has been set in a third-party context.
+
+A third-party context occurs when the subresource the cookie is being set on has a cross-site frame ancestor.
+Once a frame is considered to be in a third-party context, all requests within that frame and its child frames are also third-party and have a cross site ancestor. Similarly, once a request redirects to a cross-site URL, it is considered third-party (even if it is subsequently redirected back to a first-party request after, that subsequent first-party request is now considered an ABA request).
+
+Note: In the table below, sites A1, A2 and A3 are all first-party to each other.
+| Site frame tree |Site cookie is set on| hasCrossSiteAncestor value of cookie| Nodes that can't access the cookie|
+|---|---|--|--|
+| A1         |A1| false         | |
+| A1->A2      |A2 | false          | 
+| A1->A2->A3   |A3   | false          |  
+| A1->B       |B| true          | B|
+| A1->B->A2    |A1   | false        | B and A2
+| A1->B->A2 |B    | true          | A1 and A2 |
```

Comment:
(continued from original thread at https://github.com/w3c/webextensions/pull/581#discussion_r1651549877)

Initially I assumed that "the second column the domain of a (sub)resource in the deepest frame for which a cookie is managed".
Your clarification suggests that the actual meaning is actually, "the second column the site of the frame in the frame tree where the cookie is set".

Since the descendants of a frame have zero effect on the ancestor bit of a cookie, the behavior of `A1->B->A2 | A1` is equivalent to `A1 | A1` (already at line 17), and `A1->B->A2 | B` is equivalent to `A1->B | B` (line 20). Is there any benefit to listing an example with descendants if descendants don't impact the behavior of the cookies in upper frames?


```suggestion
```

Reply:
I'm fine removing it.